File: examples/java-api-service/src/main/java/com/example/service/UserService.java
package com.example.service;

import com.example.model.User;
import com.example.util.Validator;
import java.util.HashMap;
import java.util.Map;
import java.util.Optional;

/**
 * Service for managing users.
 */
public class UserService {

    private final Map<String, User> users = new HashMap<>();

    public User createUser(String name, String email) {
        Validator.requireNonEmpty(name, "Name");
        Validator.requireValidEmail(email);

        User user = new User(name, email);
        users.put(user.getId(), user);
        return user;
    }

    public Optional<User> findById(String id) {
        return Optional.ofNullable(users.get(id));
    }

    public boolean deleteUser(String id) {
        return users.remove(id) != null;
    }

    public User updateUser(String id, String name, String email) {
        User user = users.get(id);
        if (user == null) {
            throw new IllegalArgumentException("User not found: " + id);
        }
        if (name != null) {
            user.setName(name);
        }
        if (email != null) {
            Validator.requireValidEmail(email);
            user.setEmail(email);
        }
        return user;
    }

    public int getUserCount() {
        return users.size();
    }
}

File: examples/kotlin-api-service/src/main/kotlin/com/example/service/UserService.kt
package com.example.service

import com.example.model.User
import com.example.util.Validator

class UserService {
    private val users = mutableMapOf<String, User>()

    fun findById(id: String): User? {
        return users[id]
    }

    fun createUser(name: String, email: String): User {
        Validator.validateEmail(email)
        val id = "user-${users.size + 1}"
        val user = User(id, name, email)
        users[id] = user
        return user
    }

    fun listAll(): List<User> {
        return users.values.toList()
    }

    fun deleteUser(id: String): Boolean {
        return users.remove(id) != null
    }
}
